src/Simplify.cpp

```diff
@@ -177,37 +175,69 @@ class Simplify : public IRMutator {
                 *max_val = b.second;
                 return true;
             }
+        } else if (const Broadcast *b = e.as<Broadcast>()) {
+            return const_int_bounds(b->value, min_val, max_val);
+        } else if (const Max *max = e.as<Max>()) {
+            int64_t min_a, min_b, max_a, max_b;
+            if (const Min *min = max->a.as<Min>()) {
```

Comment:
The canonicalize part only guarantees that the RHS are non integers.
You might have this kind of Expr like in the test case: min(max(min(x, t1), t2), 5), where t1 and t2 are another min/max clamp. Then, on visit to min(x, t1), it will actually reorder t1 and x since there is another canonicalization rule that reorders min/max to be in the LHS.